src/components/asthma/helpers/asthma-data.ts

```diff
@@ -0,0 +1,343 @@
+        return computeBiometricsForDate(date, result.deviceDataPoints);
+    },
+    loadAirQualitiesForControlStatus: async function (homeAirQualityZipCode: string, workAirQualityZipCode: string): Promise<AsthmaAirQuality[]> {
+        let result = await loadAirQualities(add(new Date(), {hours: -12}), endOfToday());
+        const computeAqi = (dataPoints: DeviceDataPoint[]) => getDataPointValueAsNumber(dataPoints.sort(sortDataPointsDesc)[0]);
+        return [
+            homeAirQualityZipCode ? computeAirQuality('home', result.deviceDataPoints, computeAqi, 'establishing') : {type: 'home', status: 'not-configured'},
+            workAirQualityZipCode ? computeAirQuality('work', result.deviceDataPoints, computeAqi, 'establishing') : {type: 'work', status: 'not-configured'}
+        ];
+    },
+    loadAirQualitiesForDate: async function (date: Date): Promise<AsthmaAirQuality[]> {
+        let result = await loadAirQualities(startOfDay(date), endOfDay(date));
+        const computeAqi = (dataPoints: DeviceDataPoint[]) => {
+            let maxValue = 0;
+            dataPoints.forEach(dataPoint => {
+                let value = getDataPointValueAsNumber(dataPoint);
+                if (value > maxValue) {
+                    maxValue = value;
+                }
+            });
+            return maxValue;
+        };
+        return [
+            computeAirQuality('home', result.deviceDataPoints, computeAqi, 'not-found'),
+            computeAirQuality('work', result.deviceDataPoints, computeAqi, 'not-found')
+        ];
+    },
+    saveLogEntry: function (logEntry: AsthmaLogEntry): Promise<void> {
+        let logEntryDataPoint: PersistableDeviceDataPoint = {
+            type: 'Asthma-LogEntry',
+            identifier: logEntry.identifier,
+            observationDate: `${logEntry.identifier}T12:00:00-06:00`,
+            value: JSON.stringify(logEntry)
+        };
+        return MyDataHelps.persistDeviceData([logEntryDataPoint]);
+    },
+    loadAndClearAlertTakeover: async function (): Promise<string | undefined> {
+        let today = startOfToday();
+
+        let params: DeviceDataPointQuery = {
+            namespace: 'Project',
+            type: ['AlertTakeover'],
+            observedAfter: today.toISOString()
+        };
+
+        let result = await MyDataHelps.queryDeviceData(params);
+
+        let dataPoints = result.deviceDataPoints;
+        if (dataPoints.length === 0) return undefined;
+
+        let mostRecentDataPoint = dataPoints.sort(sortDataPointsDesc)[0];
+        if (mostRecentDataPoint.properties?.hasOwnProperty('viewed')) return undefined;
+
+        mostRecentDataPoint.properties = mostRecentDataPoint.properties ?? {};
+        mostRecentDataPoint.properties['viewed'] = formatISO(new Date());
+        MyDataHelps.persistDeviceData([mostRecentDataPoint]).then();
+
+        return mostRecentDataPoint.value;
+    },
+    loadSurveyAnswers: async function (surveyNames: string[]): Promise<SurveyAnswer[]> {
```

Comment:
This feels like it could be a generic "queryAllSurveyAnswers" like we have "queryAllDeviceData" since it's not asthma specific at all, just gets everything

Reply:
Turns out that @reneefromhold already added one for that.  I've updated this code to call that existing function.